src/api/providers/fetchers/modelCache.ts

```diff
@@ -47,64 +47,61 @@ export const getModels = async (
 	baseUrl: string | undefined = undefined,
 ): Promise<ModelRecord> => {
 	let models = memoryCache.get<ModelRecord>(router)
-
 	if (models) {
-		// console.log(`[getModels] NodeCache hit for ${router} -> ${Object.keys(models).length}`)
 		return models
 	}
 
-	switch (router) {
-		case "openrouter":
-			models = await getOpenRouterModels()
-			break
-		case "requesty":
-			// Requesty models endpoint requires an API key for per-user custom policies
-			models = await getRequestyModels(apiKey)
-			break
-		case "glama":
-			models = await getGlamaModels()
-			break
-		case "unbound":
-			// Unbound models endpoint requires an API key to fetch application specific models
-			models = await getUnboundModels(apiKey)
-			break
-		case "litellm":
-			if (apiKey && baseUrl) {
-				models = await getLiteLLMModels(apiKey, baseUrl)
-			} else {
-				models = {}
-			}
-			break
-	}
+	try {
+		switch (router) {
+			case "openrouter":
+				models = await getOpenRouterModels()
+				break
+			case "requesty":
+				// Requesty models endpoint requires an API key for per-user custom policies
+				models = await getRequestyModels(apiKey)
+				break
+			case "glama":
+				models = await getGlamaModels()
+				break
+			case "unbound":
+				// Unbound models endpoint requires an API key to fetch application specific models
+				models = await getUnboundModels(apiKey)
+				break
+			case "litellm":
+				if (!baseUrl || !apiKey) {
+					// This case should ideally be handled by the caller if baseUrl is strictly required.
+					// However, for robustness, if called without baseUrl for litellm, it would fail in getLiteLLMModels or here.
+					throw new Error("Base URL and api key are required for LiteLLM models.")
+				}
+				models = await getLiteLLMModels(apiKey || "", baseUrl)
```

Comment:
I assume the check above helps TypeScript narrow the type such that we don't need the `|| ""`:
```suggestion
				models = await getLiteLLMModels(apiKey, baseUrl)
```